lib/src/local_working_copy.rs

```diff
@@ -1353,23 +1352,16 @@ impl TreeState {
         let mut deleted_files = HashSet::new();
         let mut diff_stream = Box::pin(
             old_tree
-                .diff_stream(new_tree, matcher, &copy_records)
-                .map(
-                    |TreeDiffEntry {
-                         source: _, // TODO handle copy tracking
-                         target: path,
-                         value: diff,
-                     }| async {
-                        match diff {
-                            Ok((before, after)) => {
-                                let result =
-                                    materialize_tree_value(&self.store, &path, after).await;
-                                (path, result.map(|value| (before.is_present(), value)))
-                            }
-                            Err(err) => (path, Err(err)),
+                .diff_stream(new_tree, matcher)
+                .map(|TreeDiffEntry { path, value: diff }| async {
+                    match diff {
+                        Ok((before, after)) => {
+                            let result = materialize_tree_value(&self.store, &path, after).await;
+                            (path, result.map(|value| (before.is_present(), value)))
                         }
-                    },
-                )
+                        Err(err) => (path, Err(err)),
+                    }
+                })
```

Comment:
both sides of this map construct a pair with `(path, ...)`.  I suspect you can implement this without using `match` at all and instead use `diff.map(|(before, after)| ...)`

Reply:
actually not because of the `.await` (this is in an `async` block)